src/officehours_api/views.py

```diff
@@ -290,3 +295,48 @@ def get_queryset(self):
 class AttendeeDetail(DecoupledContextMixin, generics.RetrieveAPIView):
     queryset = Attendee.objects.all()
     serializer_class = AttendeeSerializer
+
+class ExportMeetingStartLogs(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, queue_id=None, format=None):
+        username = request.user.username
+
+        # Query from the Queue hosts table to find all the queues the user is a host of filtered also be deleted status. 
+        # The value for deleted in the model is a DateTimeField, so if the value is not None, the queue is deleted.
+        queues_user_is_in = tuple(Queue.objects.filter(hosts__in=[request.user]).values_list('id', flat=True))
+
+        logger.info(f"User {username} requested to export meeting start logs for queues {queues_user_is_in}.")
+
+        # If they specify a single queue id, check if they're actually in it and return that
+        if queue_id:
+            # Security check that they are actually in this queue
+            if queue_id not in queues_user_is_in:
+                return Response({'detail': 'You are not a host of this queue.'}, status=status.HTTP_403_FORBIDDEN)
+            else:
+                queues_user_is_in = (queue_id)
+                filename = f"meeting_start_logs_{username}_queue_{queue_id}.csv"
+        # Otherwise, get all the logs for the queues the user is a host of
+        else:
+            filename = f"meeting_start_logs_{username}.csv"
+        
+        response = HttpResponse(content_type='text/csv')
+        response['Content-Disposition'] = f'attachment; filename="{filename}"'
+
+        writer = csv.writer(response)
+        with connection.cursor() as cursor:
+            # Add a query to just get the logs for the queues_user_is_in
+            cursor.execute("SELECT * FROM meeting_start_logs where queue_id in (%s)" % queues_user_is_in)
```

Comment:
While locally testing, this works for individual queues, but for exporting all queues runs into problems with more than one. This is the error that appears on this cursor line:

> TypeError at /api/export_meeting_start_logs/
not all arguments converted during string formatting

I looked up the this error message & found a solution to rewrite the tuple: https://stackoverflow.com/a/1455623
You may need two different versions of this line based on the length of `queues_user_is_in`, so that the multiple queues case looks like this:
`cursor.execute("SELECT * FROM meeting_start_logs where queue_id in %s" % (queues_user_is_in,)) # for multiple only`

Reply:
I changed this to create placeholders and use the correct way of passing values into the execute. This looks like it's working now for single and multiple queues. I also found that the call "to all_objects" instead of just "objects" returns deleted objects as well.